module/zfs/zfs_ioctl.c

```diff
@@ -2522,11 +2522,29 @@ zfs_prop_set_special(const char *dsname, zprop_source_t source,
 	case ZFS_PROP_VOLSIZE:
 		err = zvol_set_volsize(dsname, intval);
 		break;
+	case ZFS_PROP_VOLTHREADING:
+		err = zvol_set_volthreading(dsname, intval);
+		/*
+		 * Set err to -1 to force the zfs_set_prop_nvlist code down the
+		 * default path to set the value in the nvlist.
+		 */
+		if (err == 0)
+			err = -1;
+		break;
 	case ZFS_PROP_SNAPDEV:
-		err = zvol_set_snapdev(dsname, source, intval);
+		err = zvol_set_common(dsname, ZFS_PROP_SNAPDEV, source, intval);
 		break;
 	case ZFS_PROP_VOLMODE:
-		err = zvol_set_volmode(dsname, source, intval);
+		err = zvol_set_common(dsname, ZFS_PROP_VOLMODE, source, intval);
+		break;
```

Comment:
You could combine those two cases into one `err = zvol_set_common(dsname, prop, source, intval);`